python/shotgun_model/shotgun_model.py

```diff
@@ -669,19 +671,67 @@ def _finalize_item(self, item):
         """
         Called whenever an item is fully constructed, either because a shotgun query returned it
         or because it was loaded as part of a cache load from disk.
-        
-        .. note:: You can subclass this if you want to run post processing on 
+
+        .. note:: You can subclass this if you want to run post processing on
             the data as it is arriving. For example, if you are showing a list of
-            task statuses in a filter view, you may want to remember which 
+            task statuses in a filter view, you may want to remember which
             statuses a user had checked and unchecked the last time he was running
             the tool. By subclassing this UI you can easily apply those settings
-            before items appear in the UI. 
+            before items appear in the UI.
 
-        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model. 
+        :param item: :class:`~PySide.QtGui.QStandardItem` that is about to be added to the model.
             This has been primed with the standard settings that the ShotgunModel handles.
         """
         # the default implementation does nothing
 
+    def _set_tooltip(self, item, sg_item):
+        """
+        Called when an item is constructed for the first time.
+
+        .. note:: You can subclass this if you want to set your own tooltip for the model item. By
+            default, the SG_ASSOCIATED_FIELD_ROLE data is retrieved and the field name is used to
+            determine which field to pick tooltip information from.
+            - If the field is an entity (e.g. entity), then the display name of that entity's type
+              will be used. (1)
+            - If the field is part of a sub-entity (e.g entity.Asset.sg_asset_type), the display
+              name of the sub-entity's type followed by '-' and the sub-entity's field display name
+              will be used. (2)
+            - If the field is part of an entity and not an entity field(e.g. content), the display
+              name of the entity's type will be used. (3)
+            For example:
+            {
+                "type": "Task",
+                "entity": { <-- (1) Tooltip becomes "Asset"
+                    "sg_asset_type: "Character", <-- (2) Tooltip becomes "Asset - Type"
+                    "type": "Asset",
+                    "code": "Alice"
+                },
+                "content": "Art" <-- (3) Tooltip becomes "Task"
```

Comment:
I like where you are going with this, but in your example Shotgun would actually show
Associated Sequence > Sequence > Short Name.
So, in my example above, it would become

Asset 'Alicia''
Task 'Art'
Link > Asset > Type 'Character'